algo/uidlist.go

```diff
@@ -105,34 +105,43 @@ func IntersectCompressedWithBin(dec *codec.Decoder, q []uint64, o *[]uint64) {
 
 	// Pick the shorter list and do binary search
 	if ld < lq {
-		uids := dec.Uids()
-		for len(uids) > 0 {
-			for _, u := range uids {
-				qidx := sort.Search(len(q), func(idx int) bool {
-					return q[idx] >= u
-				})
-				if qidx >= len(q) {
-					return
-				}
-				if q[qidx] == u {
-					*o = append(*o, u)
-					qidx++
-				}
-				q = q[qidx:]
+		last_q := q[len(q)-1]
+		uids := make([]uint64, 0)
+		for {
+			n := dec.Uids()
+			if len(n) == 0 {
+				break
+			}
+			uids = append(uids, n...)
```

Comment:
We should avoid creating a big list here, that is not necessary. We could just keep doing intersection with one block at a time.